### PR TITLE
Fix separator,  stat_prefix and listener.traffic_direction

### PR DESCRIPTION
**Pull request by mandarjog**
1. Use stat separator
2. Use traffic direction API

3. Use stat_prefix -- not working ...
4. Update mtls flag.

### CODE REVIEW

**Comment by mandarjog** [comment]
At present fails with 
```
[2019-08-16 17:01:25.263][1052011][info][config] [external/envoy/source/server/configuration_impl.cc:72] loading 1 listener(s)
[2019-08-16 17:01:25.278][1052011][critical][main] [external/envoy/source/server/server.cc:94] error initializing configuration 'extensions/stats/testdata/client.yaml': Protobuf message (type envoy.config.filter.http.wasm.v2.Wasm reason INVALID_ARGUMENT:stat_prefix: Cannot find field.) has unknown fields
[2019-08-16 17:01:25.278][1052011][info][main] [external/envoy/source/server/server.cc:595] exiting
Protobuf message (type envoy.config.filter.http.wasm.v2.Wasm reason INVALID_ARGUMENT:stat_prefix: Cannot find field.) has unknown fields
```

**Comment by mandarjog** [comment]
`stat_prefix` is added to `api/envoy/config/wasm/v2/wasm.proto` but not to `api/envoy/config/accesslog/v2/wasm.proto` or to `api/envoy/config/filter/http/wasm/v2/wasm.proto`

**Comment by mandarjog** [comment]
I think it needs to be added there.

**Comment by mandarjog** [comment]
I will have to update this once #2365 merges.

**Comment by mandarjog** [comment]
I am going to use `=.=` and `;.;` as fs and vs respectively. This seems like a hack, but it is only a config change.

**Comment by mandarjog** [comment]
@jplevyak Update, PTAL.

**Comment by mandarjog** [comment]
@bianpengyuan please review, esp updates to stackdriver code. Thanks.

**Comment by bianpengyuan** [comment]
will approve once John takes a look.

File: extensions/common/context.cc [hunk 1]
```diff
@@ -83,6 +102,19 @@ void populateHTTPRequestInfo(RequestInfo *request_info) {
       getHeaderMapValue(HeaderMapType::RequestHeaders, kMethodHeaderKey)
           ->toString();
   getRequestDestinationPort(&request_info->destination_port);
+
+  std::string tls_version;
+  bool cert_presented;
+
+  if (outbound) {
+    getResponsePeerCertificatePresented(&cert_presented);
```
Comment (jplevyak) [inline extensions/common/context.cc]: Currently both these are outbound.  We will need Kuat's fixes and then he may change the API as well.

File: extensions/stats/config.proto [hunk 2]
```diff
@@ -18,20 +18,22 @@ syntax = "proto3";
...
+  int32 max_peer_cache_size = 2;
+
+  // prefix to add to stats emitted by the plugin.
+  string stat_prefix = 3;  // default: "istio_"
+
+  // Stats api squashes dimensions in a single string.
+  // The squashed string is parsed at prometheus scrape time to recover
+  // dimensions. The following 2 fields set the field and value separators {key:
+  // value} -->  key{value_separator}value{field_separator}
+  string field_separator = 4;  // default: ";;"
```
Comment (jplevyak) [inline extensions/stats/config.proto]: We had talked about using ;.

Comment (jplevyak) [reply extensions/stats/config.proto]: semicolon comma

Comment (mandarjog) [reply extensions/stats/config.proto]: The actual default is `;.;` will fix this.

**Pull request merged**